core/trino-main/src/main/java/io/trino/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -403,7 +403,7 @@ public TableWriterNode map(TableWriterNode node, PlanNode source, PlanNodeId new
                 map(node.getFragmentSymbol()),
                 map(node.getColumns()),
                 node.getColumnNames(),
-                node.getNotNullColumnSymbols(),
+                map(ImmutableList.copyOf(node.getNotNullColumnSymbols())).stream().collect(toImmutableSet()),
```

Comment:
You can simplify it by using `public Symbol map(Symbol symbol)`

```suggestion
                node.getNotNullColumnSymbols().stream().map(this::map).collect(toImmutableSet()),
```

Reply:
but apart from this is it fine in your opinion ?

Reply:
The change in `SymbolMapper` is fine. 
Please change the test so that it fails before the change, according to https://github.com/trinodb/trino/pull/13462#discussion_r936383826

Reply:
Thank you! Actually I spoke with @findepi and he proposed different approach that I am currently working on.
I will ask for another review when I am done :)